src/SharpBrick.PoweredUp/Functions/DiscoverPorts.cs

```diff
@@ -124,6 +124,10 @@ private void UpdateKnowledge(byte[] data, LegoWirelessMessage message)
                 _logger?.LogInformation($"Stage: {_stageTwoCount}/{_stageTwoExpected}, Messages: {ReceivedMessages}/{SentMessages} ");
 
                 CheckEndOfDiscovery();
+            } 
+            else if (message is GenericErrorMessage)
+            {
+                ReceivedMessages++;
```

Comment:
I call that cheating ;) I spent this morning about a more semantic approach. I mean, we could correlate the response of the PortModeRequestInfo and than handle this more delicately. Have to think about it.

Reply:
Ok, yes, this is totally cheating :) I can also simply remove this, as it has nothing to do with this device's implementation really.

I agree that correlating request & response is the right way to fix this, especially since we do not know what exactly new devices will fail on and it will help if someone/users with the device can report the output including failures, instead of having no info and requiring proper debugging. And going by how these different devices behave, I guess we can assume for new devices, anything can fail :)